packages/graph/hash_graph/lib/graph/src/types/schema/link_type.rs

```diff
@@ -0,0 +1,91 @@
+use serde::{Deserialize, Serialize};
+
+use crate::types::schema::VersionedUri;
+
+/// Will serialize as a constant value `"linkType"`
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+enum LinkTypeTag {
+    LinkType,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct LinkType {
+    kind: LinkTypeTag,
+    #[serde(rename = "$id")]
+    id: VersionedUri,
+    title: String,
+    description: String,
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
```

Comment:
Is having both `default` and `skip_serializing_if = "Vec::is_empty"` redundant? I suppose `default` alone would cover both situations of empty vecs and no `related_keywords`

Reply:
`default` is for _de_serializing, and `skip_...` is for `serializing` (kudos to @TimDiekmann for explanation)